docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+```
+
+Depending on the data type, DeepReg will expect the images and labels to be stored in
+specific structures: check the [data loader configuration](dataset_loader.html) for more
+details.
+
+### Labeled key - Required
+
+The `labeled` key indicates whether labels should be used during training. A Boolean is
+used to indicate the usage of labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  labeled: true
+```
+
+### Type key - Required
+
+The type of data loader used will depend on how one wants to train the network.
+Currently, DeepReg data loaders support the "paired", "unpaired" and "grouped" training
+strategies. Passing a string that doesn't match any of the above would raise an error.
+The data loader type would be specified using the `type` key:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+```
+
+#### Data loader dependent keys
+
+Depending on which string is passed to the `type` key, DeepReg will initialise a
+different data loader instance with different sampling strategies. These are reviewed in
+depth in the [dataset loader configuration](dataset_loader.html) documentation. Here we
+outline the arguments necessary to configure the different dataloaders.
+
+###### Sample_label - Required
+
+In the case that we have more than one label per image, we need to indicate to the
+loader which one to use. We can use the `sample_label` argument to indicate which method
+to use. Pass one of "sample", for random sampling, or "all" to use all the available
+labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+  labeled: true
+  sample_label: "sample" # one of "sample", "all" or None
+```
+
+In the case the `labeled` argument is false, the sample_label is unused, but still must
+be passed. Additionally, if the tensors in the files only have one label, irregardles of
+the `sample_label` argument, the data loader will only pass the one label to the
+network.
```

Comment:
it is not clear, we have a more detailed explanation to this cite the link: https://deepreg.readthedocs.io/en/latest/docs/exp_label_sampling.html

also, @mathpluscode what is the difference between "all" and "sample" during training? - it is not clear.

Reply:
For training,
- `all` means for each image we will yield image-label pairs for all labels
- `sample` means for each image we will yield only one image-label pair

Reply:
how can you yield all pairs during training?

Reply:
random with replacement

Reply:
Ok, to clarify:
- all: for one image, x labels: yelds x image label pairs with the same image a
- sample: for one image, with x labels: yields 1 image label pair randomly sampled.

Reply:
@mathpluscode please clarify/correct this. If it is still sample one at a time, we should separately explain this for train and test, as you can't use more than one image-label pair at a time:
During training:
all: sampling without replacement
sample: sampling with replacement
During testing:
all: sampling without replacement until all labels exhausted
sample: randomly sampling one image-label pair at a test time

Reply:
@YipengHu @mathpluscode

Reply:
@NMontanaBrown - had a chat with @mathpluscode I think we can clarify this by:
"sample_label" specifies the label sampling method to construct an epoch of image-label pairs for training. 
The default for training is the "sample" option, which randomly samples one label for each image to form an epoch, resulting one image-label pair for each available image. The option "all" samples all available labels for each image, resulting in larger and unequal number of image-label pairs. The "all" is the default option during test stage predicting for all possible image-label pairs.

Reply:
So, then...
Ok, to clarify:
- all: for one image, x labels: yields x image label pairs with the same image a. Occurs over all images, in one epoch.
- sample: for one image, with x labels: yields 1 image label pair randomly sampled. Occurs for all images in one epoch?